// reactstrap-date-picker/lib/calendar/Calendar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Calendar = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactstrap = require("reactstrap");

var _setTimeToNoon = require("../util/setTimeToNoon");

var _CalendarHeader = require("./CalendarHeader");

var _CalendarSubHeader = require("./CalendarSubHeader");

var _CalendarBody = require("./CalendarBody");

var _CalendarFooter = require("./CalendarFooter");

var _useCalendarDays = require("./useCalendarDays");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Calendar = _ref => {
  var {
    selectedDate,
    displayDate,
    minDate,
    maxDate,
    onChange,
    dayLabels,
    cellPadding,
    weekStartsOn,
    showTodayButton,
    todayButtonLabel,
    roundedCorners,
    showWeeks,
    monthLabels,
    previousButtonElement,
    nextButtonElement,
    placement,
    open,
    container,
    target,
    onChangeMonth
  } = _ref;
  var calendarDays = (0, _useCalendarDays.useCalendarDays)(displayDate, selectedDate, minDate, maxDate, weekStartsOn);

  var handleDayClick = e => {
    var day = e.currentTarget.getAttribute('data-day');
    var newSelectedDate = (0, _setTimeToNoon.setTimeToNoon)(new Date(displayDate));
    newSelectedDate.setDate(day);
    onChange(newSelectedDate);
  };

  var handleTodayClick = () => {
    var newSelectedDate = (0, _setTimeToNoon.setTimeToNoon)(new Date());
    onChange(newSelectedDate);
  };

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement(_reactstrap.Popover, {
    className: "rdp-popover ".concat(placement) //toggle     = {() => handleHide()}
    ,
    isOpen: open,
    container: container,
    target: target,
    placement: placement // delay      = {200} //  does not apply for us (manual triggering)

  }, /*#__PURE__*/_react.default.createElement(_reactstrap.PopoverHeader, {
    tag: "div"
  }, /*#__PURE__*/_react.default.createElement(_CalendarHeader.CalendarHeader, {
    previousButtonElement: previousButtonElement,
    nextButtonElement: nextButtonElement,
    displayDate: displayDate,
    minDate: minDate,
    maxDate: maxDate,
    onChange: newDisplayDate => onChangeMonth(newDisplayDate),
    monthLabels: monthLabels
  })), /*#__PURE__*/_react.default.createElement(_reactstrap.PopoverBody, null, /*#__PURE__*/_react.default.createElement("table", {
    className: "rdp-calendar text-center"
  }, /*#__PURE__*/_react.default.createElement(_CalendarSubHeader.CalendarSubHeader, {
    dayLabels: dayLabels,
    showWeeks: showWeeks,
    cellPadding: cellPadding
  }), /*#__PURE__*/_react.default.createElement(_CalendarBody.CalendarBody, {
    calendarDays: calendarDays,
    showWeeks: showWeeks,
    onDayClick: handleDayClick,
    cellPadding: cellPadding,
    roundedCorners: roundedCorners
  }), /*#__PURE__*/_react.default.createElement(_CalendarFooter.CalendarFooter, {
    dayLabels: dayLabels,
    showWeeks: showWeeks,
    handleTodayClick: handleTodayClick,
    showTodayButton: showTodayButton,
    todayButtonLabel: todayButtonLabel
  })))));
};

exports.Calendar = Calendar;
Calendar.propTypes = {
  selectedDate: _propTypes.default.object,
  displayDate: _propTypes.default.object,
  //.isRequired,
  minDate: _propTypes.default.string,
  maxDate: _propTypes.default.string,
  onChange: _propTypes.default.func.isRequired,
  dayLabels: _propTypes.default.array.isRequired,
  cellPadding: _propTypes.default.string.isRequired,
  weekStartsOn: _propTypes.default.number,
  showTodayButton: _propTypes.default.bool,
  todayButtonLabel: _propTypes.default.string,
  roundedCorners: _propTypes.default.bool,
  showWeeks: _propTypes.default.bool,
  monthLabels: _propTypes.default.array.isRequired,
  previousButtonElement: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.object]).isRequired,
  nextButtonElement: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.object]).isRequired,
  placement: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.func])
};
